casper/src/test/resources/ListOpsTest.rho

```diff
@@ -61,6 +65,50 @@ in {
             ], *ackCh)
         }
       }
-    }
+    } |
+
+    contract test_any(rhoSpec, _, ackCh) = {
+      new ch0, ch1, ch2, ch3, ch4 in {
+
+        @ListOps!("any", [], *greaterThan1, *ch0) |
+        @ListOps!("any", [1], *greaterThan1, *ch1) |
+        @ListOps!("any", [2], *greaterThan1, *ch2) |
+        @ListOps!("any", [3, 4], *greaterThan1, *ch3) |
+        @ListOps!("any", [1, 2, 3, 4], *greaterThan1, *ch4) |
+
+        rhoSpec!("assertMany",
+          [
+            ((false, "== <-", *ch0), "any returns false for empty lists"),
+            ((false, "== <-", *ch1), "[1]"),
+            ((true, "== <-", *ch2), "[2]"),
+            ((true, "== <-", *ch3), "[3,4]"),
+            ((true, "== <-", *ch4), "[1,2,3,4]")
+          ], *ackCh)
+      }
+    } |
+
+    contract test_any_shortcircuit(rhoSpec, _, ackCh) = {
+      new callCountCh, greaterThan1WithCount, resultCh in {
+        callCountCh!(0) |
+        contract greaterThan1WithCount(@x, returnCh) = {
+          for (@count <- callCountCh) {
+            callCountCh!(count + 1) |
+            returnCh!(x > 1)
+          }
+        } |
+
+        @ListOps!("any", [2, 1, 0], *greaterThan1WithCount, *resultCh) |
+
+        for (@result <- resultCh) {
+          rhoSpec!("assertMany",
+            [
+              ((true, "==", result), "first item satisfies the predicate"),
+              ((1, "== <-", *callCountCh), "after the first failure no more calls should happen"),
```

Comment:
If we're checking whether there exists an element of the collection that satisfies a predicate, why should we short-circuit when one element fails? It should actually be the opposite. The call short-circuits after the first success. Similarly,`forall` should only short-circuit after the first failure.

Reply:
This is a typo in the message. The test is correct